src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleDictionary.cs

```diff
@@ -30,28 +30,9 @@ private static void HandleStartDictionary(JsonSerializerOptions options, ref Rea
                 state.Current.InitializeJsonPropertyInfo();
                 state.Current.CollectionPropertyInitialized = true;
 
-                ClassType classType = state.Current.JsonClassInfo.ClassType;
-                if (classType == ClassType.Value)
-                {
-                    Type elementClassInfoType = jsonPropertyInfo.ElementClassInfo.Type;
-                    if (elementClassInfoType != typeof(object) && elementClassInfoType != typeof(JsonElement))
-                    {
-                        ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(state.Current.JsonClassInfo.Type);
-                    }
-                }
-
                 JsonClassInfo classInfo = state.Current.JsonClassInfo;
 
-                if (state.Current.IsProcessingEnumerable())
-                {
-                    if (classInfo.CreateObject == null)
-                    {
-                        ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(classInfo.Type);
-                        return;
-                    }
-                    state.Current.ReturnValue = classInfo.CreateObject();
-                }
-                else if (state.Current.IsProcessingDictionary())
+                if (state.Current.IsProcessingDictionary())
```

Comment:
It could be, and when it is, we should throw a `JsonException` (done in the `else` branch).
Please see https://github.com/dotnet/corefx/pull/42254/files#r341406865 for reasoning.